orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/pipeline/job/PreconfiguredJobStage.groovy

```diff
@@ -95,13 +97,38 @@ class PreconfiguredJobStage extends RunJobStage {
     String[] props = mapping.split(/\./)
     Object current = root
     for (int i = 0; i < props.length - 1; i++) {
-      Object next = current[props[i]]
-      if (next == null) {
-        throw new IllegalArgumentException("no property ${props[i]} on $current")
+      Object next
+      if(props[i] ==~ /.*\[\d+\]$/) {
+        next = getValueFromArrayExpression(current, props[i])
+      } else {
+        next = current[props[i]]
+        if (next == null) {
+          throw new IllegalArgumentException("no property ${props[i]} on $current")
+        }
       }
       current = next
     }
     current[props.last()] = value
   }
 
+  private static Object getValueFromArrayExpression(Object root, String expression) {
+    // TODO: Do we need to handle arrays nested in other arrays?
+    String[] parts = expression.split(/[\[\]]/)
+    String propName = parts[0]
+    int index = -1
+    try {
+      index = Integer.parseInt(parts[1])
+    } catch (NumberFormatException ex) {
+      throw new IllegalArgumentException("Unable to parse index from expresion ${expression}", ex)
+    }
+    List<Object> nextList = root[propName]
+    if (nextList == null) {
+      throw new IllegalArgumentException("no property ${propName} on $root")
+    }
+    if (nextList.size() <= index || index < 0) {
+      throw new IllegalArgumentException("Invalid index $index for list $nextList")
```

Comment:
maybe
```suggestion
      throw new IllegalArgumentException("Invalid index $index for list $propName")
```

since that will actually have the list name